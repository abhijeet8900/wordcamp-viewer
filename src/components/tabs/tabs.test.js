import { fireEvent, render, screen } from "@testing-library/react";
import React from "react";
import Tabs from "./index.js";

describe("Tabs tests", () => {
  const MOCK_TABS = [
    {
      id: "tab-1",
      title: "Tab 1",
    },
    {
      id: "tab-2",
      title: "Tab 2",
    },
  ];

  it("Renders Tabs ", () => {
    render(<Tabs options={MOCK_TABS} />);
    MOCK_TABS.forEach((tab) => {
      const heading = screen.getByText(tab.title);
      expect(heading).toBeInTheDocument();
    });
  });

  it("Changes Tab on click ", () => {
    let currentTab = "";
    render(
      <Tabs
        options={MOCK_TABS}
        onClick={(tab) => {
          currentTab = tab;
        }}
      />
    );
    fireEvent.click(screen.getByText(MOCK_TABS[1].title));

    expect(currentTab).toBe(MOCK_TABS[1].id);
  });

  it("Snapshot test ", () => {
    const { container } = render(<Tabs options={MOCK_TABS} />);
    expect(container).toMatchSnapshot();
  });
});
