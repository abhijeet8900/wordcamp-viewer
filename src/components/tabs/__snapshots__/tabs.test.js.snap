// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs tests Snapshot test  1`] = `
<div>
  <div
    class="y w-full flex  gap-x-2 justify-evenl px-4 mb-5 "
  >
    <span
      class="w-full px-4 shadow-md border-b-2 text-black font-bold text-center cursor-pointer hover:bg-blue-400 "
    >
      Tab 1
    </span>
    <span
      class="w-full px-4 shadow-md border-b-2 text-black font-bold text-center cursor-pointer hover:bg-blue-400 "
    >
      Tab 2
    </span>
  </div>
</div>
`;
